### PR TITLE
Enabling running tests for a single assembly through build.cmd

### PR DESCRIPTION
**Pull request by Priya91**
From the commandline, build.cmd /p:input="System.Collections.Immutable" will compile all the src projects, but compiles only the system.collections.immutable.test project and runs only the immutable tests. After this commit is merged, subsequent changes will be added to corefx repository in build.cmd and build.proj

### CODE REVIEW

**Comment by mmitche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://corefx-ci.cloudapp.net/jenkins/job/BuildTools_Build/9/
Test FAILed.

**Comment by Priya91** [comment]
@FiveTimesTheFun : Please review.

File: src/Microsoft.DotNet.Build.Tasks/Targets/tests.targets [hunk 1]
```diff
@@ -9,7 +9,8 @@
 
   <Target Name="FindTestDirectories">
     <ItemGroup>
-      <AllTestAssemblies Include="$(BaseOutputPathWithConfig)*Tests\*Tests.dll" />
+      <AllTestAssemblies Condition="'$(SingleTestAssembly)' == ''" Include="$(BaseOutputPathWithConfig)*Tests\*Tests.dll" />
+	  <AllTestAssemblies Condition="'$(SingleTestAssembly)' != ''" Include="$(BaseOutputPathWithConfig)*$(SingleTestAssembly).Tests\*Tests.dll" />
```
Comment (mellinoe) [inline src/Microsoft.DotNet.Build.Tasks/Targets/tests.targets]: I'm a little concerned that we are making too many assumptions about the names of assemblies (if we weren't already). We already assume a lot about what test assemblies are named (which isn't actually always true for some of our tests), and this here seems to add the assumption that test assembly names have the product assembly name somewhere in them. In general I think this seems a bit too fragile. What do others think?

**Comment by weshaggard** [comment]
We shouldactually create another wrapper script to run tests for a single test as opposed to overloading build.cmd to run just one test. Perhaps adding a RunTests.cmd helper script.

**Comment by mmitche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://corefx-ci.cloudapp.net/jenkins/job/dotnet_buildtools_prtest/2/
Test PASSed.

**Comment by FiveTimesTheFun** [comment]
Thanks @Priya91 - I'm a bit reluctant to recommend this, I'd like to keep the "API" to the build (the available options/properties) as small as possible.  I imagine your scenario is that you're just modifying one library, and want to iteratively run tests for just that library?  Could this be achieved by reusing the xunit traits infrastructure that @krwq plumbed through?

**Comment by mellinoe** [comment]
> Could this be achieved by reusing the xunit traits infrastructure that @krwq plumbed through?

You could do something sort of similar, but you'd probably have to add a trait to all of the classes in the assembly that you want to test (like "Category', "RunThis") and then include that on the command line. Would be kind of annoying to do that for all your test classes and then remember to delete it at the end. I think Wes's suggestion is best.

**Comment by Priya91** [comment]
@weshaggard, @FiveTimesTheFun , @mellinoe  : Yes, creating a runtests.cmd seems like a good idea, instead of modifying build.cmd. 
But i dont need another target to run a single assembly tests, coz only the input is changing. So i'll change this PR to expect Test assembly inputs, and run only tests from those assemblies. Sounds good?

**Comment by Priya91** [comment]
Creating a new PR for this.

**Pull request closed without merging**